### PR TITLE
Add normalize_spaces flag to arrow.get

### PR DESCRIPTION
**Pull request by meddir**
## Pull Request Checklist

Thank you for taking the time to improve Arrow! Before submitting your pull request, please check all *appropriate* boxes:

- [x] 🧪 Added **tests** for changed code.
- [x] 🛠️ All tests **pass** when run locally (run `tox` or `make test` to find out!).
- [x] 🧹 All linting checks **pass** when run locally (run `tox -e lint` or `make lint` to find out!).
- [x] 📚 Updated **documentation** for changed code.
- [x] ⏩ Code is **up-to-date** with the `master` branch.

If you have *any* questions about your code changes or any of the points above, please submit your questions along with the pull request and we will try our best to help!

## Description of Changes

Added  normalize_spaces flag to factory.get(), parser.parse(), and parser.parse_iso(). This would remove redundant whitespace from the datetime_string so that the parsing would behave as expected.

Closes: #421

### CODE REVIEW

**Comment by systemcatch** [comment]
Hey @meddir thanks for the PR, we're all really busy right now with work/life stuff but don't worry we haven't forgotten this.

**Comment by jadchaar** [comment]
@meddir one thing that I would like to see added is documentation to alert users to this new flag: https://github.com/crsmithdev/arrow/blob/master/docs/index.rst.

File: tests/test_parser.py [hunk 1]
```diff
@@ -652,6 +652,14 @@ def test_parse_W(self):
...
                 self.parser.parse(fmt, "W")
 
+    def test_parse_normalize_spaces(self):
+        assert self.parser.parse(
+            "Jun 1 2005  1:33PM", "MMM D YYYY H:mmA", normalize_spaces=True
+        ) == datetime(2005, 6, 1, 13, 33)
+        assert self.parser.parse(
+            "  \n Jun   1\t 2005\n ", "MMM D YYYY", normalize_spaces=True
+        ) == datetime(2005, 6, 1)
+
```
Comment (systemcatch) [inline tests/test_parser.py:662]: We should add a test that checks this string will raise a parser error if `normalize_spaces=False`.

Comment (systemcatch) [reply tests/test_parser.py:662]: Also it would be good to test this without a format string.

**Comment by jadchaar** [comment]
@systemcatch I have addressed your feedback and added documentation. Mind taking another look before we merge this in?

File: docs/index.rst [hunk 2]
```diff
@@ -489,19 +489,32 @@ You can also escape regular expressions by enclosing them within square brackets
...
 
-Date formats may be fenced on either side by one punctuation character from the following list: :literal:`, . ; : ? ! " \` ' [ ] { } ( ) < >`
+Date and time formats may be fenced on either side by one punctuation character from the following list: ``, . ; : ? ! " \` ' [ ] { } ( ) < >``
 
 .. code-block:: python
 
-    >>> arrow.get("Tomorrow (2019-10-31) is Halloween!", "YYYY-MM-DD")
-    <Arrow [2019-10-31T00:00:00+00:00]>
+    >>> arrow.get("Tomorrow (2019-10-31T09:12:45) is Halloween!", "YYYY-MM-DDTHH:mm:ss")
+    <Arrow [2019-10-31T09:12:45+00:00]>
```
Comment (systemcatch) [inline docs/index.rst:497]: Not bothered either way but why is this changed?

Comment (jadchaar) [reply docs/index.rst:497]: I changed it because I noticed that the documentation only mentioned that dates can be fenced by punctuation, while both dates and times can. So I just tweaked one of the examples to include times :).
